### PR TITLE
Clip using input seek

### PR DESCRIPTION
**Pull request by miranovy**
| Q                  | A
| ------------------ | ---
| Bug fix?           | no
| New feature?       | yes
| BC breaks?         | no
| Deprecations?      | no
| Fixed tickets      | fixes #issuenum
| Related issues/PRs | #issuenum
| License            | MIT

#### What's in this PR?

Cutting video with clip filter is slow because use output seeking method. 
This PR add clip that use input seeking method. It is faster option. For more information see http://trac.ffmpeg.org/wiki/Seeking

#### Why?

Because cut filter is slow. 

#### Example Usage

```php
$clip = $video->clip(FFMpeg\Coordinate\TimeCode::fromSeconds(30), FFMpeg\Coordinate\TimeCode::fromSeconds(15));
$clip->save(new FFMpeg\Format\Video\X264(), 'video.avi');
```

On clip you can apply same filters as on video. For example resizing filter.
```php
$clip = $video->clip(FFMpeg\Coordinate\TimeCode::fromSeconds(30), FFMpeg\Coordinate\TimeCode::fromSeconds(15));
$clip->filters()->resize(new FFMpeg\Coordinate\Dimension(320, 240), FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_INSET, true);
$clip->save(new FFMpeg\Format\Video\X264(), 'video.avi');
```

### CODE REVIEW

File: src/FFMpeg/Media/Video.php [hunk 1]
```diff
@@ -315,4 +49,16 @@ public function concat($sources)
     {
         return new Concat($sources, $this->driver, $this->ffprobe);
     }
+
+    /**
+     * Return clip.
```
Comment (jens1o) [inline src/FFMpeg/Media/Video.php]: `Clips the video at the given time(s).` sounds better in my opinion :)

File: tests/Unit/Media/ClipTest.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class ClipTest extends AbstractMediaTestCase
+{
+
+    /**
+     * @dataProvider provideBuildOptions
+     */
+    public function testBuildCommand($startValue, $durationValue, $commands)
+    {
+        $configuration = $this->getMock('Alchemy\BinaryDriver\ConfigurationInterface');
```
Comment (jens1o) [inline tests/Unit/Media/ClipTest.php]: Please use `\Alchemy\BinaryDriver\ConfigurationInterface::class` (and so on) instead. :)

File: tests/Unit/Media/ClipTest.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            ->will($this->returnValue($configuration));
+
+        $ffprobe = $this->getFFProbeMock();
+
+        $start = $this->getTimeCodeMock();
+        $start->expects($this->once())
+            ->method('__toString')
+            ->will($this->returnValue($startValue));
+
+        if (is_null($durationValue)) {
```
Comment (jens1o) [inline tests/Unit/Media/ClipTest.php]: use `null !== $durationValue` instead

File: tests/Unit/Media/ClipTest.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        $clip = new Clip($this->getVideoMock(__FILE__), $driver, $ffprobe, $start, $duration);
+        $fc = $clip->getFinalCommand($format, $outputPathfile);
+
+        $this->assertCount(1, $fc);
+        $this->assertStringStartsWith(implode(' ', $commands), $fc[0]);
+    }
+
+    public function provideBuildOptions()
+    {
+        return array(
```
Comment (jens1o) [inline tests/Unit/Media/ClipTest.php]: Please use the new `[]` array notation :)

**Comment by jens1o** [comment]
Thanks for your contribution. :)

File: tests/Unit/Media/ClipTest.php (hunk 2 above)
Comment (miranovy) [reply tests/Unit/Media/ClipTest.php]: sorry I do not understand :(

File: tests/Unit/Media/ClipTest.php (hunk 3 above)
Comment (miranovy) [reply tests/Unit/Media/ClipTest.php]: replaced

File: tests/Unit/Media/ClipTest.php (hunk 2 above)
Comment (jens1o) [reply tests/Unit/Media/ClipTest.php]: Unrelated in this code branch. There is nothing to improve in this codestyle :)

**Comment by jens1o** [comment]
Thank you! I'm going to test this in a bit and then file a new release with this pull request. :)

**Comment by jens1o** [comment]
Plan to test this at the weekend. Sorry for the delay :/

**Comment by jens1o** [comment]
AWESOME! It is working fine for me! Thank you very much. 🎉 

I want to file a new release tomorrow, hope that's okay for you. :)

**Pull request merged**